src/cpu/aarch64/matmul/acl_matmul.hpp

```diff
@@ -166,17 +205,17 @@ struct acl_matmul_t : public primitive_t {
 
     status_t execute(const exec_ctx_t &ctx) const override {
         if (pd()->weights_format_kind_ == format_kind::any) {
-            return execute_forward_fixed_format(ctx);
+            return execute_forward<true>(ctx);
         } else {
-            return execute_forward_non_fixed_format(ctx);
+            return execute_forward<false>(ctx);
         }
     }
 
 private:
     // To guard the const execute_forward(), the mutex must be 'mutable'
     mutable std::mutex mtx;
-    status_t execute_forward_non_fixed_format(const exec_ctx_t &ctx) const;
-    status_t execute_forward_fixed_format(const exec_ctx_t &ctx) const;
+    template <bool IsFixedFormat>
```

Comment:
Would it be possible to convert template argument into a runtime member? From the code perspective, template usage is not quite justified.

Reply:
Sorry, this was merged in on #1889, I rebased to reflect that

Reply:
It would be really great if you could plan to code this part differently in the next release.

In general, we'd like to keep templates in kernels only, when it provides a better flexibility over splitting features not supported across different ISA. I know it's not the case as of now, but there's a motion in that direction. If we could go this direction since the beginning for ACL, that would be really great.

Reply:
Makes sense, I've made an internal issue to handle this